cryptography/hedera-cryptography-rpm/src/main/rust/raps/src/raps.rs

```
@@ -143,31 +133,16 @@ fn generate_statement(
     ab_next: &AddressBook,
     signatures: &Signatures,
     tss_vk_next_hash: [u8; HASH_LENGTH],
-) -> Result<([u8; 32], [u8; 32], Statement), ()> {
-    let ab_curr_hash = ab_rotation_lib::address_book::serialize_and_digest_sha256(&ab_curr);
-    let ab_next_hash = ab_rotation_lib::address_book::serialize_and_digest_sha256(&ab_next);
+) -> Result<([u8; HASH_LENGTH], [u8; HASH_LENGTH], Statement), RAPSError> {
+    let ab_curr_hash = ab_rotation_lib::address_book::serialize_and_digest_sha256(&ab_curr)?;
+    let ab_next_hash = ab_rotation_lib::address_book::serialize_and_digest_sha256(&ab_next)?;
 
-    let ab_prev_hash = match prev_proof {
-        Some(prev_proof) => {
-            let parsed_prev_proof = match PublicValuesStruct::abi_decode(&prev_proof.public_values.to_vec(), true) {
-                Ok(val) => val,
-                Err(_) => return Err(())
-            };
-            Some(parsed_prev_proof.ab_curr_hash.0)
-        },
-        None => None
-    };
+    let prev_proof_pub_values = prev_proof
+        .map(|p| { PublicValuesStruct::abi_decode(&p.public_values.to_vec(), true)})
+        .and_then(|p| p.ok());
```

This code isn't equivalent to the old version. Specifically, it swallows any errors from `abi_decode()` call and pretends as if we simply don't have a previous AB hash. Swallowing errors doesn't seem right to me.

Same comment applies to the `tss_vk_prev_hash` computation below as well.

---

That's true. Let me fix it.